File: java/src/main/java/com/adaptionsoft/games/uglytrivia/domain/Player.java
package com.adaptionsoft.games.uglytrivia.domain;

import com.adaptionsoft.games.uglytrivia.domain.events.GoldCoinWon;

public class Player {
    public final String name;
    private int location;
    private int goldCoins;
    private boolean isInPenaltyBox;

    public Player(String name) {
        this.name = name;
    }

    public void move(int roll) {
        location += roll;
        if (location > 11) location = location - 12;
    }

    public int getLocation() {
        return location;
    }

    public GoldCoinWon winAGoldCoin() {
        goldCoins++;
        return new GoldCoinWon(name, goldCoins);
    }

    public boolean hasPlayerWon() {
        return goldCoins == 6;
    }

    public boolean isInPenaltyBox() {
        return isInPenaltyBox;
    }

    public void sendToPenaltyBox() {
        isInPenaltyBox = true;
    }
}

File: php/src/Domain/Player.php
<?php


namespace Trivia\Domain;


use Trivia\Domain\Event\GoldCoinWon;
use Trivia\Domain\Event\PlayerMoved;
use Trivia\Domain\Event\PlayerWonGame;

class Player
{
    private $name;
    private $location = 0;
    private $goldCoins = 0;
    private $isInPenaltyBox = false;

    /**
     * Player constructor.
     * @param $name
     */
    public function __construct($name)
    {
        $this->name = $name;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getLocation(): int
    {
        return $this->location;
    }

    public function move($roll) : array
    {
        $this->location = $this->location + $roll;
        if ($this->location > 11) $this->location = $this->location - 12;

        return array(new PlayerMoved($this->name, $this->location));
    }

    public function hasPlayerWon(): bool
    {
        return $this->goldCoins == 6;
    }

    public function winAGoldCoin() : array
    {
        $this->goldCoins++;
        $events = [];
        array_push($events, new GoldCoinWon($this->name, $this->goldCoins));
        if ($this->hasPlayerWon())
            array_push($events, new PlayerWonGame($this->name));
        return $events;
    }

    public function isInPenaltyBox() : bool
    {
        return $this->isInPenaltyBox;
    }

    public function goToPenaltyBox()
    {
        $this->isInPenaltyBox = true;
    }
}